packages/mdc-chips/chip-set/foundation.ts

```diff
@@ -104,6 +111,183 @@ export class MDCChipSetFoundation extends MDCFoundation<MDCChipSetAdapter> {
     this.adapter_.removeChip(chipId);
   }
 
+  handleChipKeyboard(chipId: string, key: string) {
+    const maxIndex = this.adapter_.getChipListLength() - 1;
+    const index = this.adapter_.getIndexOfChipById(chipId);
+    let idx = -1;
+    if (HORIZONTAL_KEYS.has(key)) {
+      idx = this.determineNextHorizontalChip_(chipId, key);
+    } else if (VERTICAL_KEYS.has(key)) {
+      idx = this.determineNextVerticalChip_(chipId, key);
+    }
+
+    // Early exit if the index is invalid or the source of the event
+    if (idx < 0 || idx > maxIndex || idx === index) {
+      return;
+    }
+
+    this.adapter_.focusChipAtIndex(idx);
+  }
+
+  private determineNextVerticalChip_(chipId: string, key: string): number {
```

Comment:
Can you point me to article (a11y) that explains if chips need to support vertical focus traversal?

IMO, UP arrow & LEFT arrow have the same behaviour - i.e., changes focus to previous chip. Similarly, DOWN & RIGHT arrow focuses the next chip in the list.

Reply:
As per this example in https://www.w3.org/TR/wai-aria-practices-1.1/examples/grid/LayoutGrids.html#

The only different that I understand between DOWN key and RIGHT key is that, DOWN moves focus to next logical row (A chip containing content and remove icon) where as RIGHT moves focus to every element.

WDYT?

Reply:
Replied on Slack